examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+ *
+ */
+otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    return &sTransmitFrame;
+}
+
+/**
+ * Get the most recent RSSI measurement.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns The RSSI in dBm when it is valid.  127 when RSSI is invalid.
+ *
+ */
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    signed char rssi_cur = -110;
+    rssi_cur             = rf_get_rssi();
+    return rssi_cur;
+}
+
+/**
+ * Begin the energy scan sequence on the radio.
+ *
+ * This function is used when radio provides OT_RADIO_CAPS_ENERGY_SCAN capability.
+ *
+ * @param[in] aInstance      The OpenThread instance structure.
+ * @param[in] aScanChannel   The channel to perform the energy scan on.
+ * @param[in] aScanDuration  The duration, in milliseconds, for the channel to be scanned.
+ *
+ * @retval OT_ERROR_NONE             Successfully started scanning the channel.
+ * @retval OT_ERROR_NOT_IMPLEMENTED  The radio doesn't support energy scanning.
+ *
+ */
+otError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aScanChannel);
+    OT_UNUSED_VARIABLE(aScanDuration);
+
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+void otPlatRadioEnableSrcMatch(otInstance *aInstance, bool aEnable)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    sSrcMatchEnabled = aEnable;
+}
+
+static uint32_t m_in_critical_region = 0;
+
+static inline void util_disable_rf_irq(void)
+{
+    rf_clr_irq_mask(FLD_RF_IRQ_RX);
+    m_in_critical_region++;
+}
+
+static inline void util_enable_rf_irq(void)
+{
+    m_in_critical_region--;
+    if (m_in_critical_region == 0)
+    {
+        rf_set_irq_mask(FLD_RF_IRQ_RX);
+    }
+}
+
+static void setupTransmit(otRadioFrame *aFrame)
+{
+    int           i;
+    unsigned char rf_data_len;
+    unsigned int  rf_tx_dma_len;
+
+    rf_data_len   = aFrame->mLength - 1;
+    tx_buffer[4]  = aFrame->mLength;
+    rf_tx_dma_len = rf_tx_packet_dma_len(rf_data_len);
+    tx_buffer[3]  = (rf_tx_dma_len >> 24) & 0xff;
+    tx_buffer[2]  = (rf_tx_dma_len >> 16) & 0xff;
+    tx_buffer[1]  = (rf_tx_dma_len >> 8) & 0xff;
+    tx_buffer[0]  = rf_tx_dma_len & 0xff;
+
+    for (i = 0; i < aFrame->mLength - 2; i++)
+    {
+        tx_buffer[5 + i] = aFrame->mPsdu[i];
+    }
+
+    rf_set_channel(aFrame->mChannel);
+}
+
+phy_ccaSts_t rf_performCCA(void)
+{
+    unsigned int t1        = clock_time();
+    signed char  rssi_peak = -110;
+    signed char  rssi_cur  = -110;
+
+    while (!clock_time_exceed(t1, 128))
+    {
+        rssi_cur = rf_get_rssi();
+        if (rssi_cur > rssi_peak)
+        {
+            rssi_peak = rssi_cur;
+        }
+    }
+
+    if (rssi_peak > -60)
+    {
+        // return PHY_CCA_BUSY;
+        return PHY_CCA_IDLE;
+    }
+    else
+    {
+        return PHY_CCA_IDLE;
+    }
```

Comment:
It seems no matter the value of `rssi_peak` is below or above -60, the function will return `PHY_CCA_IDLE `;